src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/config/DataSourceElement.java

```diff
@@ -288,6 +291,16 @@ private BasicDataSource initPool(String maxPool) {
             dataSource.setPassword(getPassword());
         }
 
+        if(isPreinit()) {
+            // side effect - connection pool init - that is what we want
+            try {
+                dataSource.getConnection().close();
+                log.debug("Preinitializing the connection pool: {}@{}", getDataSourceName(), System.identityHashCode(dataSource));
+            } catch (SQLException ex) {
+                log.error("Error preinitializing the connection pool: {}@{}", getDataSourceName(), System.identityHashCode(dataSource), ex);
```

Comment:
same thing with log.isErrorEnabled()

Reply:
See my reply to log.isDebugEnabled().

Reply:
Errors should be rare, so I don't see a reason to use `isErrorEnabled` ever.

Reply:
Thanks, I also surrounded it with a check - others did it, too, in the source code, I found several examples.